### DISCUSSION TITLE
Return value of Filesystem::deleteDirectory(), deleteDirectories() and cleanDirectory()

### DISCUSSION THREAD

**Discussion by mlocati** [laravel/framework#51664] [General]
1. The `Filesystem::deleteDirectory()` method returns false if the directory does not exist, and true in any other cases.
   I think it should return false if the directory can't be deleted too.
2. `Filesystem::deleteDirectories()` returns false if the directory does not contain any sub-directories; shouldn't it return false if it can't delete these sub-directories? (and changing the return value to true if there's no directory to be deleted)
3. `Filesystem::cleanDirectory()` returns false if the directory does not exist, and true in any other cases.
   I think it should return false also if the directory hasn't been empties too.

**Comment by RehArk** [comment]
Hi, i agree with that, because i m trying to delete a directory, the response send is true but the directory still existe
